src/createServer.js

```diff
@@ -1,8 +1,89 @@
+/* eslint-disable no-console */
 'use strict';
 
+const http = require('node:http');
+const { formidable } = require('formidable');
+const fs = require('node:fs');
+const path = require('node:path');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = new http.Server();
+
+  server.on('request', async (req, res) => {
+    if (req.url === '/' || req.url === '/favicon.ico') {
+      res.on('error', (err) => {
+        console.error(err);
+        res.statusCode = 400;
+        res.writeHead(200, { 'Content-Type': 'text/plain' });
+
+        return res.end('Failed forme parse!');
+      });
+      res.writeHead(200, { 'Content-Type': 'text/html' });
+
+      res.write(`
+        <form action='/submit-expense' method="POST" style='display:flex; flex-direction:column; width:300px; justify-content:flex-end;'>
+            <label for='date'>Date:</label>
+            <input type='date' name='date' id='date' required />
+
+            <label for='title'>Title:</label>
+            <input type='text' name='title' id='title' required />
+
+            <label for='amount'>Amount:</label>
+            <input type='number' name='amount' id='amount' required />
+
+            <button type='submit'>Submit</button>
+        </form>
+    `);
+
+      return res.end('');
+    }
+
+    if (req.method === 'POST') {
+      const form = formidable({});
+
+      form.parse(req, (err, { date, title, amount }) => {
+        if (err) {
+          res.writeHead(404, { 'Content-Type': 'text/plain' });
```

Comment:
The status code here should be 500 or another appropriate error code, as 404 is typically used for 'Not Found' errors, not for form parsing errors.